cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractDistributedProgramRunner.java

```diff
@@ -171,6 +171,10 @@ protected EventHandler createEventHandler(CConfiguration cConf) {
   // returns the path of the file, if its a local URI. If it's an HDFS URI, copy it to local file system first.
   private String getKeytabPath(URI keytabURI, File tempDir) throws IOException {
     if (!"hdfs".equals(keytabURI.getScheme())) {
+      // if it's not hdfs, assume its local file
+      if (keytabURI.getScheme() != null && !"file".equals(keytabURI.getScheme())) {
+        throw new IllegalArgumentException("Currently only file:// and hdfs:// URIs are supported.");
```

Comment:
"... for keytab paths"? Would be good to mention the remedial action for this situation too.